pkg/services/ngalert/store/alert_rule.go

```diff
@@ -270,23 +269,44 @@ func (st DBstore) GetAlertRules(ctx context.Context, query *ngmodels.GetAlertRul
 	})
 }
 
-// GetNamespaces returns the folders that are visible to the user
+// GetNamespaces returns the folders that are visible to the user and have at least one alert in it
 func (st DBstore) GetNamespaces(ctx context.Context, orgID int64, user *models.SignedInUser) (map[string]*models.Folder, error) {
 	namespaceMap := make(map[string]*models.Folder)
+
+	searchQuery := models.FindPersistedDashboardsQuery{
+		OrgId:        orgID,
+		SignedInUser: user,
+		Type:         searchstore.TypeAlertFolder,
+		Limit:        -1,
+		Permission:   models.PERMISSION_VIEW,
+		Sort:         models.SortOption{},
+		Filters: []interface{}{
+			searchstore.FolderWithAlertsFilter{},
+		},
+	}
+
 	var page int64 = 1
 	for {
-		// if limit is negative; it fetches at most 1000
-		folders, err := st.FolderService.GetFolders(ctx, user, orgID, -1, page)
+		query := searchQuery
+		query.Page = page
+		proj, err := st.SQLStore.FindDashboards(ctx, &query)
```

Comment:
This might be a naive question, but it's super weird to me that `FindDashboards` returns folders, especially folders that may not contain any dashboards at all. This does work for folders which contain _only_ alert rules and no dashboards, right?

Reply:
Folders and Dashboards are historically stored in the same table. As I mentioned in the PR description this method is buried under a few abstractions (FolderService and SearchService). I was thinking about introducing a new FolderService or changing signature of FolderService to be more flexible but then decided to go with the smallest possible footprint. 

I opened an issue to figure out a better approach https://github.com/grafana/grafana/issues/47228